src/mscorlib/corefx/System/Globalization/TextInfo.cs

```diff
@@ -282,33 +268,15 @@ public unsafe virtual String ToLower(String str)
             return ChangeCase(str, toUpper: false);
         }
 
-        private static Char ToLowerAsciiInvariant(Char c)
+        static private Char ToLowerAsciiInvariant(Char c)
         {
-            Contract.Assert(IsAscii(c));
-            return (Char)s_toLowerAsciiMapping[c & 0x7F];
+            if ('A' <= c && c <= 'Z')
+            {
+                c = (Char)(c | 0x20);
+            }
+            return c;
```

Comment:
Is this approach faster?  Or it's close enough that you felt it was worth it to remove the static table?